lib/gorgon/core_ext/hash/deep_merge.rb

```diff
@@ -0,0 +1,21 @@
+# Algorithm from 'activesupport/core_ext/hash/deep_merge'
```

Comment:
If I add `activesupport` gem, it also brings `minitest` gem which causes some tests to fail.

Moreover, since we only need `deep_merge`, installing the entire `activesupport` gem is an overkill.